core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1095,7 +1095,7 @@ func (o *evmTxStore) FindTxWithSequence(ctx context.Context, fromAddress common.
 	err = o.Transact(ctx, true, func(orm *evmTxStore) error {
 		var dbEtx DbEthTx
 		err = orm.q.GetContext(ctx, &dbEtx, `
-SELECT * FROM evm.txes WHERE from_address = $1 AND nonce = $2 AND state IN ('confirmed', 'confirmed_missing_receipt', 'unconfirmed')
+SELECT * FROM evm.txes WHERE from_address = $1 AND nonce = $2 AND state IN ('confirmed', 'confirmed_missing_receipt', 'unconfirmed', 'finalized')
```

Comment:
I think we want to exclude `finalized` in this query. This is only used for the `ForceRebroadcast` method which would only be run on tx that are stuck in the mempool not yet finalized.